[Rally] Facing `panic: runtime error` while running benchmarking for data streams

- Command ran:- `elastic-package benchmark rally --benchmark container-benchmark -v`
- Error:-
```
2023/12/11 16:30:24 ERROR unable to refresh data stream at the beginning of rally run: [404 Not Found] {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [metrics-kubernetes.container-ep]","resource.type":"index_or_alias","resource.id":"metrics-kubernetes.container-ep","index_uuid":"_na_","index":"metrics-kubernetes.container-ep"}],"type":"index_not_found_exception","reason":"no such index [metrics-kubernetes.container-ep]","resource.type":"index_or_alias","resource.id":"metrics-kubernetes.container-ep","index_uuid":"_na_","index":"metrics-kubernetes.container-ep"},"status":404}
2023/12/11 16:30:24 DEBUG generating corpus data to /root/.elastic-package/tmp/rally_corpus...
2023/12/11 16:33:04 DEBUG running rally...
2023/12/11 16:33:04 DEBUG output command: /usr/bin/docker ps -a --filter label=com.docker.compose.project=elastic-package-stack --format {{.ID}}
2023/12/11 16:33:04 DEBUG output command: /usr/bin/docker inspect 2a306135026b 593bbd371aec bf16b5aaa522 298a7d4c0601 9c5c1e734621 b6bb79674069 570f56f7db29 113ba771f80b 9b1bb30d6ac7 8924e7c9c5e4
2023/12/11 16:33:04 DEBUG Configuring rally with Elasticsearch host from current profile (profile: default, host: "https://127.0.0.1:9200/")
2023/12/11 16:33:04 DEBUG output command: /usr/local/bin/esrally race --race-id=742d1a97-a308-4473-87c4-7e1e2bf4200d --report-format=csv --report-file=/root/.elastic-package/tmp/rally_corpus/report-1299584919.csv --target-hosts={"default":["https://127.0.0.1:9200/"]} --track-path=/root/.elastic-package/tmp/rally_corpus/track-1565619757.json --client-options={"default":{"basic_auth_user":"elastic","basic_auth_password":"changeme","use_ssl":true,"verify_certs":false}} --pipeline=benchmark-only --kill-running-processes
2023/12/11 16:33:05 ERROR unable to refresh data stream at the end of rally run: [404 Not Found] {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [metrics-kubernetes.container-ep]","resource.type":"index_or_alias","resource.id":"metrics-kubernetes.container-ep","index_uuid":"_na_","index":"metrics-kubernetes.container-ep"}],"type":"index_not_found_exception","reason":"no such index [metrics-kubernetes.container-ep]","resource.type":"index_or_alias","resource.id":"metrics-kubernetes.container-ep","index_uuid":"_na_","index":"metrics-kubernetes.container-ep"},"status":404}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x9a5a1d9]

goroutine 72 [running]:
github.com/elastic/elastic-package/internal/benchrunner/runners/rally.(*collector).createEventsFromMetrics(0xc02a5a0, {0x0, 0x0, 0x0})
        /opt/buildkite-agent/builds/bk-agent-prod-gcp-1702055724337323931/elastic/elastic-package/internal/benchrunner/runners/rally/metrics.go:337 +0xa9
github.com/elastic/elastic-package/internal/benchrunner/runners/rally.(*collector).start.func1()
        /opt/buildkite-agent/builds/bk-agent-prod-gcp-1702055724337323931/elastic/elastic-package/internal/benchrunner/runners/rally/metrics.go:102 +0xb7
created by github.com/elastic/elastic-package/internal/benchrunner/runners/rally.(*collector).start in goroutine 1
        /opt/buildkite-agent/builds/bk-agent-prod-gcp-1702055724337323931/elastic/elastic-package/internal/benchrunner/runners/rally/metrics.go:96 +0x89
 ```
 
cc:- @tommyers-elastic

--- a/internal/benchrunner/runners/rally/metrics.go
+++ b/internal/benchrunner/runners/rally/metrics.go
@@ -334,14 +334,16 @@ func (c *collector) createEventsFromMetrics(m metrics) [][]byte {
 
 	var nEvents []interface{}
 
-	for node, stats := range m.nMetrics.Nodes {
-		nEvents = append(nEvents, nEvent{
-			Ts:          m.ts * 1000, // ms to s
-			ClusterName: m.nMetrics.ClusterName,
-			NodeName:    node,
-			NodeStats:   &stats,
-			Meta:        c.metadata,
-		})
+	if c.startMetrics.nMetrics != nil {
+		for node, stats := range m.nMetrics.Nodes {
+			nEvents = append(nEvents, nEvent{
+				Ts:          m.ts * 1000, // ms to s
+				ClusterName: m.nMetrics.ClusterName,
+				NodeName:    node,
+				NodeStats:   &stats,
+				Meta:        c.metadata,
+			})
+		}
 	}
 
 	var events [][]byte

